src/compiler.rs

```diff
@@ -63,13 +74,24 @@ impl<F> Default for CompilationUnit<F> {
 
             columns: Default::default(),
             exposed: Default::default(),
+
             num_steps: Default::default(),
+            q_enable: Default::default(),
+            first_step: Default::default(),
+            last_step: Default::default(),
+
             num_rows: Default::default(),
 
             polys: Default::default(),
             lookups: Default::default(),
 
             fixed_assignments: Default::default(),
+
+            ast_id: Default::default(),
+            uuid: uuid(),
+
+            other_sub_circuits: Default::default(),
+            other_columns: Default::default(),
         }
     }
 }
```

Comment:
I think we can do:
```
...
uuid: uuid(),
..Default::default()
}
```

Reply:
We are implementing Default for this type, so if the compiler allows what you said it would end up in a recursive infinite loop.